reana_workflow_controller/config.py

```diff
@@ -154,10 +154,40 @@ def _env_vars_dict_to_k8s_list(env_vars):
 )
 """Common to all workflow engines environment variables for debug mode."""
 
-JUPYTER_INTERACTIVE_SESSION_DEFAULT_IMAGE = (
-    "docker.io/jupyter/scipy-notebook:notebook-6.4.5"
-)
-"""Default image for Jupyter based interactive session deployments."""
+REANA_INTERACTIVE_SESSIONS_ENVIRONMENTS = json.loads(
+    os.getenv("REANA_INTERACTIVE_SESSIONS_ENVIRONMENTS", "{}")
+)
+"""Allowed and recommended environments to be used for interactive sessions.
+
+This is a dictionary where keys are the type of the interactive session.
+For each session type, a list of recommended Docker images are provided (`recommended`)
+and whether custom images are allowed (`allow_custom`).
+
+Example:
+{
+    "jupyter": {
+        "recommended": [
+            {
+                "name": "Jupyter SciPy Notebook 6.4.5",
+                "image": "docker.io/jupyter/scipy-notebook:notebook-6.4.5"
+            }
+        ],
+        "allow_custom": true
+    }
+}
+"""
+
+REANA_INTERACTIVE_SESSIONS_RECOMMENDED_IMAGES = {
```

Comment:
Can we amend this to be ready for situations where the configuration is empty?

For example, some site may want to prevent opening of notebooks altogether by configuring:

```yaml
interactive_sessions:
  environments:
    jupyter:
      recommended:
      allow_custom: false
```

This currently leads to a traceback when starting reana-workflow-controller:

```
Traceback (most recent call last):
  File "/code/./reana_workflow_controller/app.py", line 17, in <module>
    app = create_app()
  File "/code/./reana_workflow_controller/factory.py", line 49, in create_app
    app.config.from_object("reana_workflow_controller.config")
  File "/usr/local/lib/python3.8/dist-packages/flask/config.py", line 227, in from_object
    obj = import_string(obj)
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/utils.py", line 595, in import_string
    __import__(import_name)
  File "/code/./reana_workflow_controller/config.py", line 180, in <module>
    REANA_INTERACTIVE_SESSIONS_RECOMMENDED_IMAGES = {
  File "/code/./reana_workflow_controller/config.py", line 181, in <dictcomp>
    type_: {recommended["image"] for recommended in config["recommended"]}
TypeError: 'NoneType' object is not iterable
```

This means that the REANA would be basically down if people misconfigure the recommended images section. 

Can you make the YAML value detection more robust to prevent these cases? E.g. we could use something like `config.get("recommended", [])`  and such. We should also assume that  "image" part might be misconfigured.

Reply:
Done, it should now be more robust in case of missing recommended images, missing `allow_custom`, both here and in reana-ui